dora/src/vm/annotations.rs

```diff
@@ -131,4 +132,25 @@ impl AnnotationDefinition {
             .name;
         annotation_usages.contains(name)
     }
+
+    pub fn reject_modifiers(
+        vm: &VM,
+        modifiers: &AnnotationUsages,
+        rejects: &[AnnotationDefinitionId],
+    ) {
+        for reject in rejects.iter() {
+            let reject = vm.annotations.idx(*reject).read().name;
+            for modifier in modifiers.iter() {
+                if modifier.name == reject {
+                    vm.diag.lock().report(
+                        FileId(0),
```

Comment:
We should definitely report the right filename and position here.

Reply:
Will do.

Reply:
It uses the `file_id` from the annotated element now, position should be fine, or do you want that to be done differently?

Reply:
(I added a preceding commit that adds `FileId` to `NamespaceData`, which did not have it before.)